test/integration/controller/core/queuedworkload_controller_test.go

```diff
@@ -0,0 +1,152 @@
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("QueuedWorkload controller", func() {
+	var (
+		ns              *corev1.Namespace
+		queue           *kueue.Queue
+		badQueue        *kueue.Queue
+		nonCreatedQueue *kueue.Queue
+		clusterQueue    *kueue.ClusterQueue
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-queuedworkload-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.BeforeEach(func() {
+		clusterQueue = testing.MakeClusterQueue("cq-workload").
+			Resource(testing.MakeResource(resourceGPU).
+				Flavor(testing.MakeFlavor(flavorModelA, "5").Ceiling("10").Obj()).
+				Flavor(testing.MakeFlavor(flavorModelB, "5").Ceiling("10").Obj()).Obj()).Obj()
+		gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+		queue = testing.MakeQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, queue)).To(gomega.Succeed())
+		badQueue = testing.MakeQueue("badqueue", ns.Name).ClusterQueue("badCL").Obj()
+		gomega.Expect(k8sClient.Create(ctx, badQueue)).To(gomega.Succeed())
+		nonCreatedQueue = testing.MakeQueue("nonCreatedqueue", ns.Name).Obj()
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(framework.DeleteQueue(ctx, k8sClient, queue)).To(gomega.Succeed())
+		gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed())
+	})
+
+	ginkgo.It("Should update status when workloads are created", func() {
+		cases := []struct {
+			workload  *kueue.QueuedWorkload
+			condition kueue.QueuedWorkloadCondition
+		}{
+			{
+				// nil Queue
+				workload: testing.MakeQueuedWorkload("one", ns.Name).Request(corev1.ResourceCPU, "1").Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:    kueue.QueuedWorkloadAdmitted,
+					Status:  corev1.ConditionFalse,
+					Message: fmt.Sprintf("Queue %s for workload doesn't exist", ""),
+				},
+			},
+			{
+				// Queue has not been created
+				workload: testing.MakeQueuedWorkload("two", ns.Name).Request(corev1.ResourceCPU, "1").
+					Queue(nonCreatedQueue.Name).Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:    kueue.QueuedWorkloadAdmitted,
+					Status:  corev1.ConditionFalse,
+					Message: fmt.Sprintf("Queue %s for workload doesn't exist", nonCreatedQueue.Name),
+				},
+			},
+			{
+				// ClusterQueue non existent
+				workload: testing.MakeQueuedWorkload("three", ns.Name).Request(corev1.ResourceCPU, "1").
+					Queue(badQueue.Name).Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:    kueue.QueuedWorkloadAdmitted,
+					Status:  corev1.ConditionFalse,
+					Message: fmt.Sprintf("ClusterQueue %s doesn't exist", badQueue.Spec.ClusterQueue),
+				},
+			},
+			{
+				// Request more resources than Ceiling limit
+				workload: testing.MakeQueuedWorkload("four", ns.Name).Request(corev1.ResourceCPU, "11").
```

Comment:
This should be a case for the scheduler instead.

Reply:
sure